tide_media.install

```diff
@@ -150,3 +150,81 @@ function tide_media_update_8002() {
     }
   }
 }
+
+/**
+ * Video transcript field change from text field to WYSIWYG
+ */
+function tide_media_update_8003() {
+  $database = Drupal::database();
+
+  //step1 create two temporary tables for holding data
+  $tmp_table_names = [
+    'tmp_table' => 'media__field_media_transcript',
+    'tmp_revision_table' => 'media_revision__field_media_transcript',
+  ];
+  $spec = [
+    'type' => 'text',
+    'size' => 'big',
+    'not null' => FALSE,
+  ];
+
+  foreach ($tmp_table_names as $tmp => $table_name) {
+    if ($database->schema()->tableExists($table_name)){
+      //creates two temporary tables for holding the data
+      $database->query("CREATE TABLE {$tmp} SELECT * FROM  {$table_name}");
+      $database->truncate($table_name)->execute();
+    }
+  }
+
+  //step2 update field config from string_long to text_long
+  $fields = [
+    'field_media_transcript' => [
+      'table' => 'media__field_media_transcript',
+      'revision_table' => 'media_revision__field_media_transcript',
+      'format_col' => 'field_media_transcript_format',
+    ],
+  ];
+
+  foreach ($fields as $field_name => $f) {
+    $entity_type = 'media';
+    $field_storage = FieldStorageConfig::loadByName($entity_type, $field_name);
+
+    if (is_null($field_storage)) {
+      continue;
+    }
+
+    $new_fields = [];
+    foreach ($field_storage->getBundles() as $bundle => $label) {
+      $field = FieldConfig::loadByName($entity_type, $bundle, $field_name);
+      $new_field = $field->toArray();
+      $new_field['field_type'] = 'text_long';
+      $new_fields[] = $new_field;
+      $field->delete();
+    }
+
+    $new_field_storage = $field_storage->toArray();
+    $new_field_storage['type'] = 'text_long';
+    $new_field_storage['module'] = 'text_long';
+
+    field_purge_batch(250);
+
+    FieldStorageConfig::create($new_field_storage)->save();
+
+    foreach ($new_fields as $new_field) {
+      $new_field = FieldConfig::create($new_field);
+      $new_field->save();
+    }
+
+    //step 3 data migration
+    foreach ($tmp_table_names as $tmp => $table_name) {
+      $database->schema()->addField($tmp,'field_media_transcript_format',$spec);
+      $database->update($tmp)->fields(['field_media_transcript_format'=>'rich_text'])->execute();
+      $database->insert($table_name)->from($database->select($tmp, 't')
+        ->fields('t'))->execute();
+      $query = 'UPDATE `'."$table_name" .'` SET `field_media_transcript_value` = REPLACE(`field_media_transcript_value`, \'\r\n\', \'<br />\')';;
```

Comment:
@vincent-gao line-breaks are not always `\r\n`. `\n` and `\r` are also line-breaks. Please use `php nl2br()` instead.

Reply:
Hi @sonnykt , I prefer use mysql query directly. based on the reason you metioned before, I dont want to load all values into php.
I will update the query.